src/timetable/types.ts

```diff
@@ -0,0 +1,75 @@
+import { PopoverV2TriggerType } from "../popover-v2";
+import { ResourceScheme } from "../theme";
+
+export const ROW_BAR_COLOR_SEQUENCE = [
+    "#FFE6BB",
+    "#D8EFEB",
+    "#E6EAFE",
+    "#FAE4E5",
+    "#D3EEFC",
+] as const; // Assert to be a readonly tuple
+export const ROW_CELL_GAP = 2;
+export const ROW_INTERVAL = 15;
+export type RowBarColors = (typeof ROW_BAR_COLOR_SEQUENCE)[number];
+export type CellType = "filled" | "blocked" | "default";
+
+interface EmptyContentProps {
+    description: string;
+    illustrationScheme: ResourceScheme;
+}
+
+export interface RowData {
+    id: string;
+    name: string;
+    rowCells: RowCellData[];
+    subtitle?: string | JSX.Element | undefined;
+    rowHeaderCustomPopover?: CustomPopoverProps;
+    // HH:mm format, defaults to minTime
+    rowMinTime?: string;
+    // HH:mm format, defaults to maxTime
+    rowMaxTime?: string;
+}
+
+export interface CustomPopoverProps {
+    trigger: PopoverV2TriggerType;
+    content: string | JSX.Element;
+    width?: string | undefined;
+    padding?: string | undefined;
+    delay?: { open: number; close: number } | undefined;
+    offset?: number | undefined;
+}
+
+export interface TimeTableProps {
+    date: string;
+    emptyContent: EmptyContentProps;
+    rowData: RowData[];
+    isLoading: boolean;
+    // HH:mm format
+    minTime?: string;
+    // HH:mm format
+    maxTime?: string;
+    minDate?: string | undefined;
+    maxDate?: string | undefined;
+    totalRecords?: number | undefined;
+    width?: string | undefined;
+    height?: string | undefined;
+    outsideOpHoursCellCustomPopover?: CustomPopoverProps | undefined;
```

Comment:
this was one of my concerns with automatically computing slots. for any additional rendering behaviour, new props need to be introduced for these special slots